drivers/gpio/gpio_pcf857x.c

```diff
@@ -148,21 +158,23 @@ static int pcf8574_port_get_raw(const struct device *dev, gpio_port_value_t *val
  *
  * @param dev A pointer to the device structure
  * @param mask A mask of bits to set some bits to LOW or HIGH
- * @param value The value which is written via i2c to the pfc8574's output pins
+ * @param value The value which is written via i2c to the pcf857x's output pins
  * @param toggle A way to toggle some bits with xor
  *
  * @retval 0 If successful.
  * @retval Negative value for error code.
  */
-static int pcf8574_port_set_raw(const struct device *dev, uint8_t mask, uint8_t value,
-				uint8_t toggle)
+static int pcf857x_port_set_raw(const struct device *dev, uint16_t mask, uint16_t value,
+				uint16_t toggle)
 {
-	const struct pcf8574_drv_cfg *drv_cfg = dev->config;
-	struct pcf8574_drv_data *drv_data = dev->data;
+	const struct pcf857x_drv_cfg *drv_cfg = dev->config;
+	struct pcf857x_drv_data *drv_data = dev->data;
 	int rc = 0;
-	uint8_t tx_buf;
+	uint16_t tx_buf;
+	uint8_t tx_buf_p[2];
 
 	if (k_is_in_isr()) {
+		LOG_ERR("ewouldblock");
```

Comment:
I would remove this log, otherwise LGTM. I haven't tested this on hardware yet, but I don't want to block this MR.